backend/src/main/java/de/samuelgesang/backend/crawls/CrawlController.java

```diff
@@ -0,0 +1,63 @@
+package de.samuelgesang.backend.crawls;
+
+import de.samuelgesang.backend.sitemaps.SitemapService;
+import de.samuelgesang.backend.sites.Site;
+import de.samuelgesang.backend.sites.SiteService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/crawls")
+public class CrawlController {
+
+    private final SiteService siteService;
+    private final SitemapService sitemapService;
+    private final CrawlRepository crawlRepository;
+
+    public CrawlController(SiteService siteService, SitemapService sitemapService, CrawlRepository crawlRepository) {
+        this.siteService = siteService;
+        this.sitemapService = sitemapService;
+        this.crawlRepository = crawlRepository;
+    }
+
+    @GetMapping("/start/{siteId}")
+    public ResponseEntity<String> crawlSiteById(@PathVariable String siteId, @AuthenticationPrincipal OAuth2User user) {
+        try {
+            if (user == null) {
+                return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User is not authenticated");
```

Comment:
Ich glaube das brauchst du gar nicht. Das legst du doch in der `SecurityConfig`fest, welche Endpunkte "gesichert" sind?